### PR TITLE
Homework4: Shevchuk Aleksei

### PR DESCRIPTION
**Pull request by alex-shef**

### CODE REVIEW

File: Homework4/task2.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# 2. Города
```
Comment (belovmd) [inline Homework4/task2.py:1]: what about Brest in Belarus and France?

File: Homework4/task2.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# Дан список стран и городов каждой страны. Затем даны названия городов.
+# Для каждого города укажите, в какой стране он находится.
+
+dct = {}
+for i in range(int(input("Enter the number of lists: "))):
```
Comment (belovmd) [inline Homework4/task2.py]: where do we use i? 
for _ in ..

File: Homework4/task2.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    country, *cities = input("Enter a list: ").split()
+    for city in cities:
+        dct[city] = country
+
+for i in range(int(input("Enter the number of cities: "))):
```
Comment (belovmd) [inline Homework4/task2.py]: same issue

File: Homework4/task3.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# 3. Даны два списка чисел. Посчитайте, сколько различных чисел
+# содержится одновременно как в первом списке, так и во втором.
+
+lst1 = input("Enter the 1st list of numbers separated by a space: ").split()
+lst2 = input("Enter the 2nd list of numbers separated by a space: ").split()
+print("Number of different numbers:", len(set(lst1) | set(lst2)))
```
Comment (belovmd) [inline Homework4/task3.py]: it doesn't solve task - we need other operator

File: Homework4/task5.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# 5. Языки
+# Каждый из N школьников некоторой школы знает Mi языков.
+# Определите, какие языки знают все школьники и языки,
+# которые знает хотя бы один из школьников.
+
+print("Enter the input data, through 'Enter': ", sep='\n')
+pupils = [{input() for j in range(int(input()))} for i in range(int(input()))]
```
Comment (belovmd) [inline Homework4/task5.py]: please add invitation string

**Review by belovmd — CHANGES_REQUESTED** [review]
Please take a look at the next issues:

File: Homework4/task2.py (hunk 1 above)
Comment (alex-shef) [reply Homework4/task2.py:1]: fixed

File: Homework4/task2.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+# Для каждого города укажите, в какой стране он находится.
+
+dct = {}
+for the_list in range(int(input("Enter the number of lists: "))):
+    country, *cities = input("Enter a list: ").split()
+    dct[country] = cities
+for city in range(int(input("Enter the number of cities: "))):
+    the_city = input("Enter the city: ")
+    count = 0
+    for country, cities in dct.items():
```
Comment (belovmd) [inline Homework4/task2.py:12]: O(n*m) solution (n - num of requested cities m - size of dict)
in fact we miss dict advantages here

**Review by belovmd — COMMENTED** [review]
8,5

**Pull request closed without merging**